hydromt/data_adapter/dataframe.py

```diff
@@ -162,20 +164,21 @@ def to_file(
             )
         except IndexError as err:  # out of bounds for time
             logger.warning(str(err))
-            return None, None
+            return None, None, None
 
         if driver is None or driver == "csv":
             # always write as CSV
             driver = "csv"
             fn_out = join(data_root, f"{data_name}.csv")
             obj.to_csv(fn_out, **kwargs)
+            kwargs["index_col"] = obj.index.name
```

Comment:
The kwargs in this method are used for writing a file (here csv) and not necessarily the same as for reading the same file. See different arguments in e.g. `pandas.DataFrame.to_csv` and `pandas.read_csv`. The kwargs that are returned here are later used to read back the same file and parse it into an identical python object (here pands.DataFrame). I would suggest to return a new dictionary with `read_kwargs`, which you can initialize at the start of the `to_file` method. Idem for the other DataAdapters. I agree with your approach to start with the read_kwargs that we know are required. We can easily add once we have this infrastructure.


```suggestion
       read_kwargs = dict()
        if driver is None or driver == "csv":
            # always write as CSV
            driver = "csv"
            fn_out = join(data_root, f"{data_name}.csv")
            obj.to_csv(fn_out, **kwargs)
            read_kwargs["index_col"] = obj.index.name
```